extensions/investspec/investspec.py

```diff
@@ -0,0 +1,418 @@
+import docutils
+import importlib
+
+INPUT_TYPES_HTML_FILE = 'input_types.html'
+# accepted geometries for a vector will be displayed in this order
+GEOMETRY_ORDER = [
+    'POINT',
+    'MULTIPOINT',
+    'LINESTRING',
+    'MULTILINESTRING',
+    'POLYGON',
+    'MULTIPOLYGON']
+
+
+def format_type_string(arg_type):
+    """Represent an arg type as a user-friendly string.
+
+    Args:
+        arg_type (str): the type to format
+
+    Returns:
+        formatted string that links to a description of the input type
+    """
+    # Represent the type as a string. Some need a more user-friendly name.
+    # we can only use standard docutils features here, so no :ref:
+    # this syntax works to link to a section in a different page, but it
+    # isn't universally supported and depends on knowing the built page name.
+    if arg_type == 'freestyle_string':
+        return f'`text <{INPUT_TYPES_HTML_FILE}#text>`__'
+    elif arg_type == 'option_string':
+        return f'`option <{INPUT_TYPES_HTML_FILE}#option>`__'
+    elif arg_type == 'boolean':
+        return f'`true/false <{INPUT_TYPES_HTML_FILE}#truefalse>`__'
+    else:
+        return f'`{arg_type} <{INPUT_TYPES_HTML_FILE}#{arg_type}>`__'
+
+
+def format_units_string(unit):
+    """Represent a pint Unit as a user-friendly phrase.
+
+    Args:
+        unit (pint.Unit): the unit to format
+
+    Returns:
+        string describing the unit
+    """
+    # pluralize the first unit so that it reads more naturally
+    custom_unit_plurals = {
+        'foot': 'feet',
+        'degree_Celsius': 'degrees_Celsius'
+    }
+    units_string = str(unit)
+    if units_string == 'none':
+        return ''
```

Comment:
All args spec units are `pint.Unit`s, and that's enforced in the args spec tests. Currently there's a custom unit `pint.UnitRegistry().none` that's used for unitless numbers. I didn't use `None`, partly to enforce that every number has to have a documented unit (even if that's "unitless"), and partly because of a conceptual thing about units:

I don't fully understand it, but in some of our cases, "unitless" numbers can be thought of as having a "counting unit", which is sort of the root of all units. When we are counting something: "10 people", it's different in a sense from a pure number, like a factor in an equation. [This article](https://www.semanticarts.com/quantities-number-units-and-counting/) gives a somewhat confusing explanation, where they call the counting unit an "each". `pint` has an [equivalent unit](https://github.com/hgrecco/pint/blob/abe384033bffa7627387eeaec297f0a75777b65d/pint/default_en.txt#L112) that they call a "count". This wasn't a useful distinction to make in invest, but it's wacky to think about, and made me lean away from using `None`.